File: 4312-LimitOccurrencesInSortedArray/4312-LimitOccurrencesInSortedArray.scala
// Last updated: 20/07/2026, 10:38:27
object Solution {
    def limitOccurrences(nums: Array[Int], k: Int): Array[Int] = {
        var i = 0
        for (n <- nums) {
            if (i < k || n != nums(i - k)) {
                nums(i) = n
                i += 1
            }
        }
        nums.take(i)
    }
}

File: 4312-LimitOccurrencesInSortedArray/4312-LimitOccurrencesInSortedArray.cpp
// Last updated: 20/07/2026, 10:38:19
class Solution {
public:
    vector<int> limitOccurrences(vector<int>& nums, int k) {
        int n=0;
        for(int i:nums){
            if(n<k || i!=nums[n-k]){
                nums[n]=i;
                n+=1;
            }
        }
        nums.resize(n);
        return nums;
    }
};